### PR TITLE
enhancement: add genesis type

### PR DESCRIPTION
**Pull request by shiqizng**
adding genesis type

### CODE REVIEW

File: types/genesis_test.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+)
+
+func TestEncodeDecodeGenesis(t *testing.T) {
+	genesisStr := "{\n  \"alloc\": [\n    {\n      \"addr\": \"7777777777777777777777777777777777777777777777777774MSJUVU\",\n      \"comment\": \"RewardsPool\",\n      \"state\": {\n        \"algo\": 125000000000000,\n        \"onl\": 2\n      }\n    },\n    {\n      \"addr\": \"LX6VRFCL4723LRO3VEX5XNOXHFXFQKATKDLDBT5HNT5DFDC3SQG4EI2LSI\",\n      \"comment\": \"Wallet1\",\n      \"state\": {\n        \"algo\": 1000000000000000,\n        \"onl\": 1,\n        \"sel\": \"XGIljovdG8IjSQJRcvc/vi6UXXTIddUgjZACWZPW23M=\",\n        \"vote\": \"zccp6OB5EOh21WMV7A7iVg+CitiZeOao+VYbIkNdPgM=\",\n        \"voteKD\": 10000,\n        \"voteLst\": 3000000\n      }\n    }\n  ],\n  \"fees\": \"A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE\",\n  \"id\": \"v1\",\n  \"network\": \"sandnet\",\n  \"proto\": \"future\",\n  \"rwd\": \"7777777777777777777777777777777777777777777777777774MSJUVU\"\n}"
+	var genesis Genesis
+	json.Decode([]byte(genesisStr), &genesis)
+	assert.Equal(t, base64.StdEncoding.EncodeToString(json.Encode(genesis)), base64.StdEncoding.EncodeToString([]byte(genesisStr)))
+}
+
+func TestGenesis_Balances(t *testing.T) {
```
Comment (shiqizng) [inline types/genesis_test.go:20]: test copied from go-algorand

File: types/genesis_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncodeDecodeGenesis(t *testing.T) {
+	genesisStr := "{\n  \"alloc\": [\n    {\n      \"addr\": \"7777777777777777777777777777777777777777777777777774MSJUVU\",\n      \"comment\": \"RewardsPool\",\n      \"state\": {\n        \"algo\": 125000000000000,\n        \"onl\": 2\n      }\n    },\n    {\n      \"addr\": \"LX6VRFCL4723LRO3VEX5XNOXHFXFQKATKDLDBT5HNT5DFDC3SQG4EI2LSI\",\n      \"comment\": \"Wallet1\",\n      \"state\": {\n        \"algo\": 1000000000000000,\n        \"onl\": 1,\n        \"sel\": \"XGIljovdG8IjSQJRcvc/vi6UXXTIddUgjZACWZPW23M=\",\n        \"vote\": \"zccp6OB5EOh21WMV7A7iVg+CitiZeOao+VYbIkNdPgM=\",\n        \"voteKD\": 10000,\n        \"voteLst\": 3000000\n      }\n    }\n  ],\n  \"fees\": \"A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE\",\n  \"id\": \"v1\",\n  \"network\": \"sandnet\",\n  \"proto\": \"future\",\n  \"rwd\": \"7777777777777777777777777777777777777777777777777774MSJUVU\"\n}"
+	var genesis Genesis
+	json.Decode([]byte(genesisStr), &genesis)
+	assert.Equal(t, base64.StdEncoding.EncodeToString(json.Encode(genesis)), base64.StdEncoding.EncodeToString([]byte(genesisStr)))
+}
```
Comment (winder) [inline types/genesis_test.go:18]: Could you generate a hash on algod and confirm that you get the same hash here?

File: types/genesis.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+	// RewardsPool is the address of the rewards pool.
+	RewardsPool string `codec:"rwd"`
+
+	// FeeSink is the address of the fee sink.
+	FeeSink string `codec:"fees"`
+
+	// Timestamp for the genesis block
+	Timestamp int64 `codec:"timestamp"`
+}
```
Comment (Eric-Warehime) [inline types/genesis.go:39]: We need these too for parity with go-algorand.
```
Comment string codec:"comment"`

DevMode bool `codec:"devmode"`
```

Comment (shiqizng) [reply types/genesis.go:49]: updated.

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**